scheduler/controllers/odigosconfiguration/odigosconfiguration_controller.go

```diff
@@ -69,6 +72,28 @@ func (r *odigosConfigurationController) Reconcile(ctx context.Context, _ ctrl.Re
 		mergeRemoteConfig(&odigosConfiguration, remoteConfig)
 		logger.V(1).Info("Merged remote config into effective config")
 	}
+	// this section is for dynamic odiglet
+	// we should be able to identify the waspEnabled from the node details dynamically
+	var wasEnabled *bool // nil by default
+
+	waspEnabled := odigosConfiguration.WaspEnabled
+	// user not explicitly set the waspEnabled, so we need to check in NodeDetails
+	if waspEnabled == nil {
+		// list all NodeDetails and use the first one's WaspRequired field
+		nodeDetailsList := &odigosv1.NodeDetailsList{}
+		if err := r.List(ctx, nodeDetailsList); err != nil {
+			return ctrl.Result{}, err
+		}
+
+		// if at least one NodeDetails is found, use the first one's WaspRequired value
+		// this under the assumption that all NodeDetails have the same WaspRequired value
+		if len(nodeDetailsList.Items) > 0 {
+			wasEnabled = boolPtr(nodeDetailsList.Items[0].Spec.WaspRequired)
+			if wasEnabled != nil {
+				odigosConfiguration.WaspEnabled = wasEnabled
+			}
+		}
+	}
```

Comment:
consider [using a cm to capture dynamic detection config](https://github.com/odigos-io/odigos/blob/e9d6a9a5b74aa6242d6a010a638ed91959086f71/scheduler/controllers/odigosconfiguration/manager.go#L44), so the flow is less cryptic and more traceable